src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

```diff
@@ -963,8 +961,6 @@ public void WaitForStatus(ServiceControllerStatus desiredStatus, TimeSpan timeou
       {
         if (DateTime.UtcNow - start > timeout)
           throw new System.ServiceProcess.TimeoutException(SR.Format(SR.Timeout, ServiceName));
-
-        _waitForStatusSignal.WaitOne(250);
```

Comment:
You still need some kind of wait here, to avoid wasting 100% CPU time.

Reply:
Yeah, I thought the same.
So Thread.Sleep(250) would correct here?

Reply:
In <https://github.com/dotnet/corefx/pull/627/files#r24128155>, Thread.Sleep was called a "legacy contract", but I think it is still the best thing to use here… unless you want to go all the way and call [NotifyServiceStatusChange](https://docs.microsoft.com/windows/win32/api/winsvc/nf-winsvc-notifyservicestatuschangew), which does not look easy; I'm not even sure how to make .NET do an alertable wait so that the asynchronous procedure call can be dequeued.

Reply:
If I understand correctly, WaitHandle.WaitOne can call SynchronizationContext.Current.Wait, while Thread.Sleep cannot. So if an application is using a custom SynchronizationContext, then it can detect whether ServiceController uses WaitHandle.WaitOne or Thread.Sleep.

Reply:
Yes, this should be `Thread.Sleep(250);`.

Reply:
OK @stephentoub. Thanks for confirmation! I will get this added with next commit!